sdk/typescript/src/signature-mpc/sign.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+import type { DWalletClient } from '../client/index.js';
+import type { Ed25519Keypair } from '../keypairs/ed25519/index.js';
+import {
+	createPartialUserSignedMessages,
+	decrypt_user_share,
+	getDwalletByObjID,
+} from './dwallet_2pc_mpc_ecdsa_k1_module.js';
+import {
+	approveAndSign,
+	getEncryptedUserShare,
+	getEncryptedUserShareByObjectId,
+} from './dwallet.js';
+import { generatePaillierKeyPairFromSuiKeyPair } from './utils.js';
+
+/**
+ * Signs the given message with the given DWallet ID.
+ *
+ * @param keypair The Sui keypair that encrypted the given dwallet to itself in the past. This keypair is
+ * either the one who created the dwallet with the {@link createDWallet} function, or the one who accepted
+ * it with the {@link acceptUserShare} function.
+ */
+export const signWithDWalletID = async (
+	client: DWalletClient,
+	keypair: Ed25519Keypair,
+	dwalletID: string,
+	message: Uint8Array,
+) => {
```

Comment:
Please seperate this to two functions as the underlying are: one to presign, one to sign